bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/VelbusColorChannel.java

```diff
@@ -0,0 +1,233 @@
+
+    private int brightness = 100;
+    private int[] color = { 255, 255, 255 };
+    private int white = 100;
+
+    /**
+     * @param brightness the brightness to set
+     */
+    public void setBrightness(int brightness) {
+        this.brightness = (brightness < BRIGHTNESS_MIN_VALUE) ? BRIGHTNESS_MIN_VALUE : brightness;
+        this.brightness = (brightness > BRIGHTNESS_MAX_VALUE) ? BRIGHTNESS_MAX_VALUE : brightness;
+    }
+
+    /**
+     * @param brightnessState the brightness to set
+     */
+    public void setBrightness(PercentType brightnessState) {
+        this.brightness = brightnessState.intValue();
+    }
+
+    /**
+     * @param brightnessState the brightness to set
+     */
+    public void setBrightness(HSBType brightnessState) {
+        setBrightness(brightnessState.getBrightness());
+    }
+
+    /**
+     * @param brightness the brightness to set from velbus packet
+     */
+    public void setBrightness(byte brightness) {
+        if (brightness != VALUE_UNCHANGED) {
+            this.brightness = (Byte.toUnsignedInt(brightness) * BRIGHTNESS_MAX_VALUE)
+                    / Byte.toUnsignedInt(DALI_MAX_VALUE);
+        }
+    }
+
+    /**
+     * @return the brightness
+     */
+    public int getBrightness() {
+        return this.brightness;
+    }
+
+    /**
+     * @return the brightness in PercentType format
+     */
+    public PercentType getBrightnessPercent() {
+        return new PercentType(getBrightness());
+    }
+
+    /**
+     * @return the brightness for velbus packet
+     */
+    public byte getBrightnessVelbus() {
+        return (this.brightness == BRIGHTNESS_MAX_VALUE) ? DALI_MAX_VALUE
+                : Integer.valueOf((this.brightness * Byte.toUnsignedInt(DALI_MAX_VALUE)) / BRIGHTNESS_MAX_VALUE)
+                        .byteValue();
+    }
+
+    /**
+     * @param rgb the color to set in RGB
+     */
+    public void setColor(int[] rgb) {
+        if (rgb.length == 3) {
+            this.color[0] = (rgb[0] < COLOR_MIN_VALUE) ? COLOR_MIN_VALUE : rgb[0];
+            this.color[0] = (rgb[0] > COLOR_MAX_VALUE) ? COLOR_MAX_VALUE : rgb[0];
+            this.color[1] = (rgb[1] < COLOR_MIN_VALUE) ? COLOR_MIN_VALUE : rgb[1];
+            this.color[1] = (rgb[1] > COLOR_MAX_VALUE) ? COLOR_MAX_VALUE : rgb[1];
+            this.color[2] = (rgb[2] < COLOR_MIN_VALUE) ? COLOR_MIN_VALUE : rgb[2];
+            this.color[2] = (rgb[2] > COLOR_MAX_VALUE) ? COLOR_MAX_VALUE : rgb[2];
+        }
+    }
+
+    /**
+     * @param hsbState the color to set in HSB
+     */
+    public void setColor(HSBType hsbState) {
+        this.color = ColorUtil.hsbToRgb(hsbState);
+    }
+
+    /**
+     * @param rgb the color to set in RGB from velbus packet
+     */
+    public void setColor(byte[] rgb) {
+        if (rgb.length == 3) {
+            if (rgb[0] != VALUE_UNCHANGED) {
+                this.color[0] = (rgb[0] == DALI_MAX_VALUE) ? COLOR_MAX_VALUE : Byte.toUnsignedInt(rgb[0]);
```

Comment:
Brightness is limited from 0 to 100 in OpenHAB and use 0 to 254 value on the VMBDALI. 
We lose granularity, but during my tests I didn't see a problem with that.

In RGB, each base color (red, green, blue) use 0 to 255 value, so I use the 254 values of the VMBDALI to make it correspond.
It's only when the R/G/B color equal 254 that I change it to 255. (Because value 255 = no change).
During my tests the colors displayed on the LED strip was almost the one that I had selected on the color picker,
and the values read on the VMBDALI (on each channel) set correctly the color picker to the corresponding color.

Reply:
> Brightness is limited from 0 to 100 in OpenHAB and use 0 to 254 value on the VMBDALI.

Hm, my experience is that these sorts of assumed limits can sometimes be ignored in OpenHAB, unfortunately. For instance `commandDescription` can specify different limits for an item, or one could make a mistake and send a limit outside of the permitted range from a rule script. But then again, my familiarity with OH’s internals is limited and I might be spewing BS right now :)

But even if that was *not* the case I still think having a single function that's used everywhere all the same (even in places where the conversion is largely a no-op) would be somewhat more future proof. For instance, consider what might happen if there was a need to change the value of one of these `*_MAX_VALUE` constants. With a `rescale`-like function the code would continue working perfectly fine.

Not to mention that it is simply easier to review a single implementation of the `rescale` function :)

Reply:
rescale function implemented in last commit.